File: apps/ios/Barns/BarnsApp/Domain/UseCases/Care/GetCareTasksUseCase.swift
/// Returns the user's local care tasks.
struct GetCareTasksUseCase {
    private let repository: CareRepository

    init(repository: CareRepository) {
        self.repository = repository
    }

    func execute() async throws -> [CareTask] {
        try await repository.careTasks()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/care/GetCareTasksUseCase.kt
package com.barns.app.domain.usecase.care

import com.barns.app.domain.model.CareTask
import com.barns.app.domain.repository.CareRepository

/**
 * Returns the user's local care tasks.
 */
class GetCareTasksUseCase(
    private val repository: CareRepository,
) {
    suspend fun execute(): List<CareTask> = repository.careTasks()
}
